pkgs/development/ocaml-modules/gd4o/default.nix

```diff
@@ -0,0 +1,33 @@
+{ lib, stdenv, fetchurl, ocaml, gd, freetype, findlib, zlib, libpng, libjpeg }:
+
+stdenv.mkDerivation rec {
+  pname = "gd4o";
+  version = "1.0a5";
+
+  src = fetchurl {
+    url = "mirror://sourceforge/gd4o/gd4o/1.0%20Alpha%205/gd4o-1.0a5.tar.gz";
+    sha256 = "1vbyakz7byvxmqf3hj68rw15b4kb94ppcnhvmjv38rsyg05bc47s";
+  };
+
+  buildInputs = [ ocaml findlib libjpeg libpng ];
+  propagatedBuildInputs = [ gd zlib freetype ];
+
+  createFindlibDestdir = true;
```

Comment:
```suggestion
  preInstall = ''
    mkdir -p $OCAMLFIND_DESTDIR/stublibs
  '';
```

Reply:
I'd be happy to, but could you explain why?

Reply:
To enable the hook defined there: https://github.com/NixOS/nixpkgs/blob/f947895d6deaafbf64a80ff9ad52b31c76f7f89e/pkgs/development/tools/ocaml/findlib/default.nix#L35-L37

It will properly set the environment variable that is used by bytecode programs to locate the non-OCaml components of this library.